ERCS/erc-7738.md

```diff
@@ -0,0 +1,154 @@
+status: Draft
+type: Standards Track
+category: ERC
+created: 2024-07-01
+requires: 173
+---
+## Abstract
+
+This EIP provides a means to create a standard registry for locating executable scripts associated with the token.
+
+## Motivation
+
+[ERC-5169](./eip-5169.md) provides a client script lookup method for contracts. This requires the contract to have implemented the `ERC-5169` interface at the time of construction (or allow an upgrade path).
+
+This proposal outlines a contract that can supply prototype and certified scripts. The contract would be a multichain singleton instance that would be deployed at identical addresses on supported chains.
+
+### Overview
+
+The registry contract will supply a set of URI links for a given contract address. These URI links point to script programs that can be fetched by a wallet, viewer or mini-dapp.
+
+The pointers can be set permissionlessly using a setter in the registry contract.
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY” and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+The contract MUST implement the `IERC7738` interface.
+The contract MUST emit the `ScriptUpdate` event when the script is updated.
+The contract SHOULD order the `scriptURI` returned so that the `ERC-173` `owner()` of the contract's script entries are returned first (in the case of simple implementations the wallet will pick the first `scriptURI` returned).
+The contract SHOULD provide a means to page through entries if there are a large number of scriptURI entries.
+
+```solidity
+interface IERC7738 {
+    /// @dev This event emits when the scriptURI is updated, 
+    /// so wallets implementing this interface can update a cached script
+    event ScriptUpdate(address indexed contractAddress, string[] newScriptURI);
+
+    /// @notice Get the scriptURI for the contract
+    /// @return The scriptURI
+    function scriptURI(address contractAddress) external view returns (string[] memory);
+
+    /// @notice Update the scriptURI 
+    /// emits event ScriptUpdate(address indexed contractAddress, scriptURI memory newScriptURI);
+    function setScriptURI(address contractAddress, string[] memory scriptURIList) external;
+}
+```
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+## Rationale
+
+This method allows contracts written without the [ERC-5169](./eip-5169.md) interface to associate scripts with themselves, and avoids the need for a centralised online server, with subsequent need for security and the requires an organisation to become a gatekeeper for the database.
+
+## Test Cases
+
+Hardhat contract and test scripts for two implementations can be found in the [Asset folder](../assets/eip-7738/)
```

Comment:
Don't forget. to update this

Reply:
Yes, I'm not sure what to update to - should I remove the link, let the PR succeed, then raise a new PR to add the link in because now the assets are in the main repo?